src/pages/workspace/withPolicyConnections.tsx

```diff
@@ -27,18 +29,19 @@ function withPolicyConnections<TProps extends WithPolicyConnectionsProps>(Wrappe
         const [hasConnectionsDataBeenFetched, {status}] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY_HAS_CONNECTIONS_DATA_BEEN_FETCHED}${props.policy?.id ?? '0'}`, {
             initWithStoredValues: false,
         });
+        const isConnectionDataFetchNeeded = !isOffline && props.policy && props.policy.areConnectionsEnabled && hasConnectionsDataBeenFetched && props.policy.connections;
```

Comment:
```suggestion
        const isConnectionDataFetchNeeded = !isOffline && props.policy && props.policy.areConnectionsEnabled && !props.policy.connections && hasConnectionsDataBeenFetched === null;
```

hasConnectionsDataBeenFetched is compared with null so we don't end up making infinite requests in case the request failed (hasConnectionsDataBeenFetched value is false). This also help avoid making multiple requests because we opmisitically set the value to false.